include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to setup Google Service');
+            throw new GoogleSyncException('Unable to setup Google Service', GoogleSyncException::GSERVICE_FAILURE);
+        }
+
+        if (!$this->setUsersGoogleCalendar()) {
+            // TODO: It shouldn't be in the log as it is an exception, it will be logged if unhandled by suitecrm core.
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to setup Google Calendar Id');
+            throw new GoogleSyncException('Unable to setup Google Calendar Id', GoogleSyncException::GCALENDAR_FAILURE);
+        }
+        return true;
+    }
+
+    /**
+     * Retrieve List of meetings owned by the Current Working User
+     *
+     *
+     * @return array Array of SuiteCRM Meeting Beans
+     * @throws GoogleSyncException if $this->workingUser->id is invalid
+     * @throws GoogleSyncException if unable to get Meetings bean
+     */
+    protected function getUserMeetings($userId)
+    {
+        $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Id ' . $userId);
+        // Validate the workingUser id
+        $isValidator = new SuiteValidator();
+        if (!$isValidator->isValidId($userId)) {
+            throw new GoogleSyncException('Invalid ID requested in getUserMeetings', GoogleSyncException::INVALID_USER_ID);
+        }
+
+        // We do it this way so we also get deleted meetings
+        $query = "SELECT id FROM meetings WHERE assigned_user_id = '" . $userId . "' AND date_start <= now() + interval 3 month";
+        $result = $this->db->query($query);
+
+        $meetings = array();
+        while ($row = $this->db->fetchByAssoc($result)) {
+            $meeting = BeanFactory::getBean('Meetings');
+            if (!$meeting) {
+                throw new GoogleSyncException('Unable to get Meetings bean.', GoogleSyncException::UNABLE_TO_RETRIEVE_MEETING);
+            }
+            $meeting->retrieve($row['id'], true, false);
+            $meetings[] = $meeting;
+        }
+        return $meetings;
+    }
+
+    /**
+     * Set user's google calendar id
+     *
+     *
+     * @return bool|int Success/Failure calendar ID if success
+     */
+    protected function setUsersGoogleCalendar()
+    {
+
+        // Make sure we have a Google Calendar Service instance
+        if (!$this->isServiceExists()) {
+            return false;
+        }
+
+        // get list of users calendars
+        $calendarList = $this->gService->calendarList->listCalendarList();
+
+        // find the id of the 'SuiteCRM' calendar ... in the future, this will set the calendar of the users choosing.
+        $this->calendarId = $this->getSuiteCRMCalendar($calendarList);
+
+        // if the SuiteCRM calendar doesn't exist... Create it!
+        if (!$this->isCalendarExists()) {
+            // TODO: It should be an exception?
```

Comment:
@gymad 
I don't think it should be an exception... if it doesn't exist, it gets created. Now down at "// Final Check" we may want to throw something. I'll look into it.

Reply:
Done